packer-provisioner-goss.go

```diff
@@ -126,20 +134,30 @@ func (p *Provisioner) Prepare(raws ...interface{}) error {
 		p.config.Arch = "amd64"
 	}
 
+	if p.config.TargetOs == "" {
+		p.config.TargetOs = Linux
+	}
+
 	if p.config.URL == "" {
-		p.config.URL = fmt.Sprintf(
-			"https://github.com/aelsabbahy/goss/releases/download/v%s/goss-linux-%s",
-			p.config.Version, p.config.Arch)
+		p.config.URL = p.getDownloadUrl()
 	}
 
 	if p.config.DownloadPath == "" {
+		os := strings.ToLower(string(p.config.TargetOs))
 		if p.config.URL == "" {
-			p.config.DownloadPath = fmt.Sprintf("/tmp/goss-%s-linux-%s", p.config.Version, p.config.Arch)
+			p.config.DownloadPath = fmt.Sprintf("/tmp/goss-%s-%s-%s", p.config.Version, os, p.config.Arch)
 		} else {
 			list := strings.Split(p.config.URL, "/")
-			arch := strings.Split(list[len(list)-1], "-")[2]
+
+			file := strings.Split(list[len(list)-1], "-")
+			arch := file[2]
+			if p.isGossAlpha() {
+				// includes .exe for windows
+				arch = file[3]
```

Comment:
the comment confused me at first (I thought maybe this was only to support windows with the`.exe` - please strike it or clarify that the format of the alpha file is different.